pkg/stacks/truncate/truncate.go

```diff
@@ -84,14 +87,15 @@ func Truncate(str string, length, suffixLength int) (string, error) {
 	// See the import comment regarding use of sha1
 	// nolint:gosec
 	checksum := sha1.Sum([]byte(str))
-	retained := length - suffixLength
+	retainedLength := length - suffixLength
+	retained := strings.TrimRight(str[0:retainedLength], ".")
```

Comment:
are there other chars we need to worry about? `-` or others?

Reply:
@jbw976 
This is the regex that the specific issue is colliding against:

`[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*`

Based on that it seems that as long as periods are immediately followed by any alphanumeric it is alright, it was the lack of an alphanumeric prior to the dash causing the reported issue